lib/eslint.js

```diff
@@ -265,13 +272,29 @@ function modifyConfigsFromComments(ast, config, reportingConfig) {
                         break;
 
                     case "eslint":
-                        var items = parseJsonConfig(value);
-                        Object.keys(items).forEach(function(name) {
-                            var ruleValue = items[name];
-                            if (typeof ruleValue === "number" || (Array.isArray(ruleValue) && typeof ruleValue[0] === "number")) {
-                                commentRules[name] = ruleValue;
-                            }
-                        });
+                        try {
+
```

Comment:
Maybe move all this info parseJsonConfig? This is the only case that does more than a single function call.